sdk/data/azappconfig/CHANGELOG.md

```diff
@@ -7,6 +7,7 @@
 
 ### Breaking Changes
 * Response types `ListRevisionsPage` and `ListSettingsPage` now have the suffix `Response` in their names.
+* The optional `value *string` parameter in methods `AddSetting` and `SetSetting` has been moved to the respective options types.
```

Comment:
This is correct, but I wonder if, in practice, this is a hassle since you're unlikely to want to Add or Set a setting with a null value.

Reply:
I see your point.  The problem here is that this directly conflicts with our design guidelines and looks odd compared to other APIs.  Perhaps as a compromise we can take `value` by value and if it's the empty string we omit it.

@JeffreyRichter do you recall if this was discussed at some point?  I didn't find an APIView with any conversation history.

Reply:
Empirically, the answer appears to be yes.  Need to confirm with the service team though.

Reply:
Well, actually I think this means that we just take it be value and we do nothing special on the client.

Reply:
I will say though that when retrieving such a key, the value comes back as the empty string and not a `nil` field.

Reply:
I don't like this change. AddSetting should take a key AND a value as formal parameters.

Reply:
Value is optional.  Do we leave it as pointer-to-type then?

Reply:
Per offline discussion will leave this param as-is and document the behavior.